src/Driver/Driver.jl

```diff
@@ -340,10 +335,24 @@ function init(;
     init_driver::Bool = true,
     keyword_args...,
 )
+    # `Settings` contains global defaults
+    global_defaults = Dict{Symbol, Any}(
+        (n, getproperty(Settings, n)) for n in propertynames(Settings)
+    )
+
+    # keyword arguments must be applicable to `Settings`
     all_args = Dict{Symbol, Any}(keyword_args)
+    for kwarg in keys(all_args)
+        if get(global_defaults, kwarg, nothing) === nothing
+            @warn("Unrecognized keyword argument '$(string(kwarg))'; ignoring")
```

Comment:
Maybe it's a good idea to error instead of warn? I don't really have a preference either way, but if users configure their simulation and get results when they actually wanted a different sim, that could be a problem.

Reply:
Are there any disadvantages to making this return an error instead of a warning?

Reply:
I think an error would be better as well.  For custom CI arguments you can pass in the defaults with the argparse settings so I don't see a need to not strictly validate the arguments

Reply:
I'm now making it throw an `ArgumentError` but am happy to do something else if anyone has a suggestion.